src/Altinn.AccessManagement.Persistence/Extensions/ActivityExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+using System.Diagnostics;
+using Npgsql;
+using OpenTelemetry.Trace;
+
+namespace Altinn.AccessManagement.Persistence.Extensions;
+
+/// <summary>
+/// Extension for Activity used in Altinn.AccessManagement.Persistence
+/// </summary>
+public static class ActivityExtensions
+{
+    /// <summary>
+    /// Sets Activity tags based on command
+    /// </summary>
+    /// <param name="activity">Current activity</param>
+    /// <param name="command">NpgsqlCommand</param>
+    public static void AddSqlTags(this Activity activity, NpgsqlCommand command)
+    {
+        activity?.SetTag("db.system", "Postgres");
+        activity?.SetTag("db.statement", command.CommandText);
+
+        foreach (var param in command.Parameters.ToList())
+        {
+            try
+            {
+                if (IsList(param.Value))
```

Comment:
if (param.Value is IList params && params != null)
{
activity?.SetTag(param.ParameterName, string.Join(',', params));
}

is this pattern possible?

Reply:
To avoid calling IsList? Could unwrap to something like this

`if (param.Value == null) { continue; }
if (param.Value is IList)
{
    activity?.SetTag(param.ParameterName, string.Join(',', (IList)param.Value));
}
else
{
    activity?.SetTag(param.ParameterName, param.Value);
}`

Not sure how the parameter is stored in NpgsqlCommand, therefor the try catch. This might also be unnecessary with the .AddNpgsql() [Beta] in Programs.cs where it should pick up parameters.